shared_vcx/src/maybe_known.rs

```diff
@@ -0,0 +1,13 @@
+use serde::{Deserialize, Serialize};
+
+/// Enum used to encapsulate `string-like` types which may
+/// have variants we haven't implement yet.
+///
+/// Deserialization will be first attempted to the [`MaybeKnown::Known`] variant
+/// and then, if that fails, to the [`MaybeKnown::Unknown`] variant.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(untagged)]
+pub enum MaybeKnown<T, U = String> {
+    Known(T),
+    Unknown(U),
+}
```

Comment:
Some tests would be nice. Previously the `Protocol` type was used for this, but it be better to keep the tests contained to this crate and just make up some enums in the test module.

Reply:
The tests couldn't be moved to the new location, because there's a cyclic dependency introduced because of `Protocol`.

Reply:
Right, which is why the way to go would be creating one or two enums in the tests module only, one that takes advantage of the `String` default and maybe one that uses something else for the `Unknown` variant.